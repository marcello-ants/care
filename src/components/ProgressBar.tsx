import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import LinearProgress from '@material-ui/core/LinearProgress';

const ProgressWithStyles = withStyles((theme) => ({
  root: {
    height: 3,
    borderRadius: 5,
    marginBottom: -3,
  },
  colorPrimary: {
    backgroundColor: theme.palette.grey[theme.palette.type === 'light' ? 200 : 700],
  },
  bar: {
    borderRadius: 5,
    backgroundColor: theme.palette.care?.navy[600],
  },
}))(LinearProgress);

interface ProgressBarProps {
  stepNumber: number;
  totalSteps: number;
}

const ProgressBar = ({ stepNumber, totalSteps }: ProgressBarProps) => {
  return <ProgressWithStyles value={(stepNumber / totalSteps) * 100} variant="determinate" />;
};

export default React.memo(ProgressBar);
