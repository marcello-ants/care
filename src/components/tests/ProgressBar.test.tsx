import { render } from '@testing-library/react';
import ProgressBar from '../ProgressBar';

describe('ProgressBar', () => {
  it('should render snapshot', () => {
    const { asFragment } = render(<ProgressBar stepNumber={1} totalSteps={5} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
