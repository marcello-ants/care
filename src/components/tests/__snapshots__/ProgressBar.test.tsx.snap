// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render snapshot 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="20"
    class="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-1 MuiLinearProgress-colorPrimary WithStyles(ForwardRef(LinearProgress))-colorPrimary-2 MuiLinearProgress-determinate"
    role="progressbar"
  >
    <div
      class="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-3 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
      style="transform: translateX(-80%);"
    />
  </div>
</DocumentFragment>
`;
